ni_measurementlink_service/_internal/parameter/serializer.py

```diff
@@ -158,7 +158,9 @@ def _get_overlapping_parameters(
             )
         type = parameter_metadata_dict[field_index].type
         is_repeated = parameter_metadata_dict[field_index].repeated
-        decoder = serialization_strategy.Context.get_decoder(type, is_repeated)
+        decoder = serialization_strategy.Context.get_decoder(
+            type, is_repeated, parameter_metadata_dict[field_index].message_type
+        )
```

Comment:
One of these things is not like the others. (Two are assigned to local variables and the third is not.)

`field_metadata = parameter_metadata_dict[field_index]` would eliminate the duplicated lookups.